packages/web3/src/address/address.test.ts

```diff
@@ -17,194 +17,316 @@ along with the library. If not, see <http://www.gnu.org/licenses/>.
+      }
+      const p2mpkhAddress = bs58.encode(new Uint8Array([0x01, 0x03, ...bytes0, ...bytes1, ...bytes2, 0x02]))
+      expect(groupOfAddress(p2mpkhAddress)).toBe(groupOfLockupScript(p2mpkh))
 
-    const p2sh: LockupScript = { kind: 'P2SH', value: bytes0 }
-    const p2shAddress = bs58.encode(new Uint8Array([0x02, ...bytes0]))
-    expect(groupOfAddress(p2shAddress)).toBe(groupOfLockupScript(p2sh))
+      const p2sh: LockupScript = { kind: 'P2SH', value: bytes0 }
+      const p2shAddress = bs58.encode(new Uint8Array([0x02, ...bytes0]))
+      expect(groupOfAddress(p2shAddress)).toBe(groupOfLockupScript(p2sh))
 
-    const p2c: LockupScript = { kind: 'P2C', value: bytes0 }
-    const p2cAddress = bs58.encode(new Uint8Array([0x03, ...bytes0]))
-    expect(groupOfAddress(p2cAddress)).toBe(groupOfLockupScript(p2c))
+      const p2c: LockupScript = { kind: 'P2C', value: bytes0 }
+      const p2cAddress = bs58.encode(new Uint8Array([0x03, ...bytes0]))
+      expect(groupOfAddress(p2cAddress)).toBe(groupOfLockupScript(p2c))
+    })
   })
 
-  it('should extract token id from addresses', () => {
-    expect(binToHex(tokenIdFromAddress('wCTeteGBeSEC54GpkS8jWBzYiYNTBUuTW3WzxGd9yExT'))).toBe(
-      '25469eb0d0d0a55deea832924547b7b166c70a3554fe321e81886d3c18f19d64'
-    ),
-      expect(binToHex(tokenIdFromAddress('xrY8dxgVm38QCQXhiUFcivFutLFUNMoo8qu8vYf7wJps'))).toBe(
-        '3de370f893cb1383c828c0eb22c89aceb13fa56ddced1848db27ce7fa419c80c'
-      ),
-      expect(() => tokenIdFromAddress('eBrjfQNeyUCuxE4zpbfMZcbS3PuvbMJDQBCyk4HRHtX4')).toThrow(
-        'Invalid contract address type: 2'
-      ),
-      expect(() => tokenIdFromAddress('..')).toThrow('Non-base58 character')
+  describe('groupOfPrivateKey', () => {
+    it('should calculate the group of private key', () => {
+      expect(groupOfPrivateKey('a5c91afebe25e1644e9023e0d341ca713b59beca36e6635de37ef0c9c8689654')).toEqual(3)
+      expect(groupOfPrivateKey('82e1f50a8e372933ab8afc6362934b693e5aaa4ca308b0aac27fbe8755e0a3fa')).toEqual(2)
+    })
+  })
+
+  describe('publicKeyFromPrivateKey', () => {
+    it('should compute public key from private key', () => {
+      expect(publicKeyFromPrivateKey('91411e484289ec7e8b3058697f53f9b26fa7305158b4ef1a81adfbabcf090e45')).toBe(
+        '030f9f042a9410969f1886f85fa20f6e43176ae23fc5e64db15b3767c84c5db2dc'
+      )
+    })
   })
 
-  it('should calculate the group of private key', () => {
-    expect(groupOfPrivateKey('a5c91afebe25e1644e9023e0d341ca713b59beca36e6635de37ef0c9c8689654')).toEqual(3)
-    expect(groupOfPrivateKey('82e1f50a8e372933ab8afc6362934b693e5aaa4ca308b0aac27fbe8755e0a3fa')).toEqual(2)
-    expect(groupOfPrivateKey('860ef6d3468e93aee997617e3595859fbab634111b88a79099a7e359f447cc2f')).toEqual(2)
-    expect(groupOfPrivateKey('b3bda653c36f99ad5a88401ea99cbc2a3c25e74d3df652f1545439ec929bcf31')).toEqual(0)
+  describe('addressFromPublicKey', () => {
+    it('should compute address from public key', () => {
```

Comment:
I suggest using `PrivateKeyWallet.Random()` to create pubic/private key pair for all tests involving the use of public key and private key, so we don't need to hardcoded ones.